### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by abramsymons**

### CODE REVIEW

File: db/Dockerfile [hunk 1]
```diff
@@ -1,66 +1,52 @@
...
+    mkdir /docker-entrypoint-initdb.d && \
+    cd /tmp                                && \
     wget ${ARANGO_SIGNATURE_URL}           && \
     wget ${ARANGO_PACKAGE_URL}             && \
     gpg --verify ${ARANGO_PACKAGE}.asc     && \
     ar x ${ARANGO_PACKAGE} data.tar.gz     && \
     tar -C / -x -z -f data.tar.gz          && \
     sed -ri \
-        -e 's!^(file\s*=).*!\1 -!' \
+        -e 's!127\.0\.0\.1!0.0.0.0!g' \
```
Comment (adamstallard) [inline db/Dockerfile:30]: Does this override any firewall and expose ports to the internet?

File: web_services/foxx/brightid4/initdb.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+// deprecated collections should be added to this array after releasing
+// second update to allow 2 last released versions work together 
+const deprecated = [
+  'removed',
+  'newGroups',
+  'usersInNewGroups',
+];
+
+function createCollections() {
+  console.log("creating collections if they not exists ...");
```
Comment (adamstallard) [inline web_services/foxx/brightid4/initdb.js:24]: "if they don't exist"

Comment (abramsymons) [reply web_services/foxx/brightid4/initdb.js]: fixed, thanks

File: db/Dockerfile (hunk 1 above)
Comment (abramsymons) [reply db/Dockerfile]: You're right!
This is original Dockerfile from arango team for version 3.6.4
https://github.com/arangodb/arangodb-docker/blob/official/alpine/3.6.4/Dockerfile
This line is removed now.

File: web_services/foxx/brightid4/db.js [hunk 3]
```diff
@@ -514,6 +522,18 @@ function linkContextId(id, context, contextId, timestamp){
     throw 'contextId is duplicate';
   }
 
+  if (!userHasVerification(verification, id)) {
+    throw 'user is not verified for this context';
+  }
```
Comment (adamstallard) [inline web_services/foxx/brightid4/db.js:527]: This 
```suggestion
```

Let's remove this check to improve the user experience (so they don't go to the app and follow a deep link and have it fail).

Could you please also find and remove the check that requires someone to be verified to be sponsored. We don't want that for the same reason.  I put a long comment about it keybase://chat/brightid#node/4193

File: web_services/foxx/brightid4/db.js [hunk 4]
```diff
@@ -504,7 +512,7 @@ function userHasVerification(verification, user){
 }
 
 function linkContextId(id, context, contextId, timestamp){
-  const { collection, idsAsHex } = getContext(context);
+  const { collection, idsAsHex, verification } = getContext(context);
```
Comment (adamstallard) [inline web_services/foxx/brightid4/db.js:515]:
```suggestion
  const { collection, idsAsHex } = getContext(context);
```

File: web_services/foxx/brightid4/tests/verifications.js [hunk 5]
```diff
@@ -61,26 +67,35 @@ describe('verifications', function () {
...
-    db.linkContextId('2', 'testContext', 'new', 25);
-  });
-  context('linkContextId()', function(){
-    it('should throw "contextId is duplicate" for not duplicate contextId', function(){
+  context('linkContextId()', function() {
+    it('should not be able to link contextId to not verified user', function() {
+      (() => {
+        db.linkContextId('1', 'testContext', 'used', 5);
+      }).should.throw('user is not verified for this context');
+    });
```
Comment (adamstallard) [inline web_services/foxx/brightid4/tests/verifications.js:75]:
```suggestion
```

File: web_services/foxx/brightid4/tests/verifications.js [hunk 6]
```diff
@@ -36,6 +36,12 @@ describe('verifications', function () {
         totalSponsorships: 1
       } IN ${contextsColl}
     `;
+    query`
+      INSERT {
+        _key: "1",
+        verifications: []
+      } IN ${usersColl}
+    `;
```
Comment (adamstallard) [inline web_services/foxx/brightid4/tests/verifications.js:44]:
```suggestion
```

File: web_services/foxx/brightid4/tests/verifications.js [hunk 7]
```diff
@@ -61,26 +67,35 @@ describe('verifications', function () {
...
       db.getUserByContextId(contextIdsColl, 'testContextId').should.equal('3');
     });
+    it('should not be able to link new contextId before 8 hours', function(){
+      (() => {
+        db.linkContextId('3', 'testContext', 'testContextId2', 15);
+      }).should.throw('one contextId can be linked each 8 hours');
+    });
+    it('should be able to link new contextId after 8 hours', function(){
+      db.linkContextId('3', 'testContext', 'testContextId2', 8*3600*1000 + 15);
+    });
```
Comment (adamstallard) [inline web_services/foxx/brightid4/tests/verifications.js:93]: This test will need to be fixed to check for 3 links in 24 hours

File: web_services/foxx/brightid4/db.js (hunk 3 above)
Comment (abramsymons) [reply web_services/foxx/brightid4/db.js]: We didn't have anything for sponsor and you removed that check before for sponsoring users.
This condition is removed for linking too.

File: web_services/foxx/brightid4/tests/verifications.js (hunk 5 above)
Comment (abramsymons) [reply web_services/foxx/brightid4/tests/verifications.js]: This should be removed too now

File: web_services/foxx/brightid4/tests/verifications.js (hunk 7 above)
Comment (abramsymons) [reply web_services/foxx/brightid4/tests/verifications.js:84]: This test is updated for 3 links every day

File: web_services/foxx/brightid4/tests/verifications.js [hunk 8]
```diff
@@ -83,13 +83,13 @@ describe('verifications', function () {
...
         db.linkContextId('3', 'testContext', 'testContextId2', 15);
-      }).should.throw('one contextId can be linked each 3 days');
+      }).should.throw('one contextId can be linked each 8 hours');
     });
-    it('should be able to link new contextId after 3 days', function(){
-      db.linkContextId('3', 'testContext', 'testContextId2', 3*24*3600*1000 + 15);
+    it('should be able to link new contextId after 8 hours', function(){
+      db.linkContextId('3', 'testContext', 'testContextId2', 8*3600*1000 + 15);
     });
   });
```
[the remark this answers is not in the document]
Comment (abramsymons) [reply web_services/foxx/brightid4/tests/verifications.js]: Your changes applied
Thanks

**Pull request merged**